libs/clients/smartsolutions/src/lib/smartSolutions.types.ts

```diff
@@ -1,88 +1,68 @@
-import { Pass, PassTemplate } from '../../gen/schema'
-
-export interface VerifyPassResult {
-  valid: boolean
-  data?: string
-  name?: string
-  nationalId?: string
-  photo?: string
-  error?: PkPassVerifyError
-}
-
-export interface VerifyPassResponse {
-  valid: boolean
-  data?: {
-    updateStatusOnPassWithDynamicBarcode: Pass
-  }
+export type ServiceResponse<ResponseType> = {
+  data?: ResponseType
   errors?: {
     message: string
     path: string
   }[]
 }
 
-export interface ListPassesResponse {
-  data?: {
-    passes?: ListPassesDTO
+export type Result<ResultType, ErrorType = ServiceError> =
+  | {
+      ok: true
+      data: ResultType
+    }
+  | {
+      ok: false
+      error: ErrorType
+    }
+/*
+function ui(result: Result<Array<PassTemplate>>) {
+  if (result.ok && result.data.length > 0) {
+    // til skírteini
+  } else if (result.ok && result.data.length === 0) {
+    // á ekki skírteini
+  } else if (!result.ok) {
+    // ökusk á cut-off date fyrir mynd
+    if (result.error.canBeDisplayedToUser) {
+      return 'of gamalt ökusk'
+    } else {
+      return 'villa kom upp'
+    }
   }
 }
+*/
 
-export interface PassTemplatesDTO {
-  data: Array<PassTemplate>
-}
-
-export interface ListPassesDTO {
-  data: Array<Pass>
-}
+export type FetchResponse<T> =
+  | {
+      data: T
+      error?: never
+    }
+  | {
+      data?: never
+      error: ServiceError
+    }
 
-export interface ListTemplatesResponse {
-  data?: PassTemplate[]
+export interface ServiceError {
+  serviceCode: ErrorCode
   message?: string
-  status?: number
-}
-export interface PassTemplateDTO {
-  passTemplate: {
-    id: string
-    name: string
+  httpStatus?: {
+    status: number
+    statusText: string
   }
 }
 
-export interface UpsertPkPassResponse {
-  data: {
-    upsertPass: Pass
-  }
-  message?: string
-  status?: number
-  // If the payload is invalid, smart solutions returns a 200 ok with an errors field.
-  errors?: {
-    message: string
-    path: string
-  }[]
-}
-
-export interface PkPassVerifyError {
-  /**
-   * HTTP status code from the service.
-   * Needed while `status` was always the same, use `serviceError.status` for
-   * error reported by API.
-   */
-  statusCode: number
-  serviceError?: PkPassServiceErrorResponse
-}
-
-export interface PkPassServiceErrorResponse {
-  message?: string
-  status?: PkPassServiceVerifyPassStatusCode
-  data?: unknown
-}
-
-export type PkPassServiceVerifyPassStatusCode =
-  /** License OK */
+export type ErrorCode =
+  /** Missing Pass Template id */
   | 1
-  /** License expired */
+  /** Fetch failed */
   | 2
-  /** No license info found */
+  /** JSON parse failed */
   | 3
-  /** Request contains some field errors */
+  /** External service error */
   | 4
-  /** Unknown error */
+  /** Incomplete service response */
+  | 5
+  /** Request contains some field errors */
+  | 6
+  /** Generic error code */
```

Comment:
I'm pretty sure this is a breaking change against the app, we can't mix the `VerifyPassStatusCode` with a new error code

Reply:
Yeah, this was typed up while developing, so will fix in a second run through

Reply:
The return objects will change as well so the app will have to update anyway